app/src/main/java/de/westnordost/streetcomplete/quests/building_entrance_reference/AddEntranceReference.kt

```diff
@@ -0,0 +1,126 @@
+import de.westnordost.streetcomplete.data.quest.NoCountriesExcept
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BLIND
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.quests.building_entrance_reference.AddEntranceReferenceForm
+import de.westnordost.streetcomplete.quests.building_entrance_reference.EntranceAnswer
+import de.westnordost.streetcomplete.quests.building_entrance_reference.FlatRange
+import de.westnordost.streetcomplete.quests.building_entrance_reference.ReferenceCode
+import de.westnordost.streetcomplete.quests.building_entrance_reference.ReferenceCodeAndFlatRange
+import de.westnordost.streetcomplete.quests.building_entrance_reference.Unsigned
+
+class AddEntranceReference : OsmElementQuestType<EntranceAnswer> {
+
+    private val buildingFilter by lazy { """
+        ways with
+          building = apartments
+          and ruins != yes and disused != yes
+    """.toElementFilterExpression() }
+
+    override val changesetComment = "Add entrance identification"
+    override val wikiLink = "Key:ref"
+    override val icon = R.drawable.ic_quest_door_address
+    override val questTypeAchievements = listOf(CITIZEN, BLIND)
+    override val enabledInCountries = NoCountriesExcept(
+        "PL", // Poland - own knowledge of Mateusz Konieczny https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-879447168
+        "RU", // Russia - https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-880231076
+        "MK", // North Macedonia https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-1158016740
+        "BG", // Bulgaria - https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-879318531
+        "UA", // Ukraine - https://t.me/osmUA/123011
+        "BY", // Belarus - https://t.me/byosm/59969
+        "EE", // Estonia - https://t.me/byosm/59976
+        "HU", // Hungary https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-1159088130
+        "RO", // Romania https://t.me/OSMRomania/7363
+
+        // not applicable (no matching address system)
+        // Kosovo (XK) -  https://t.me/osmkosovo/19219
+        // Serbia (RS) - https://t.me/osm_sr/6877
+        // Lithuania (LT) - https://lists.openstreetmap.org/pipermail/talk-lt/2022-June/thread.html#2723 https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-1147833339
+        // Slovakia (SK) https://groups.google.com/g/osm_sk/c/IqZz0-iRK5I
+        // Czech Republic (CZ) https://t.me/openstreetmapcz/1242
+        // Croatia https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-1157858833
+
+        // was unable to reach communities or get replies from Kazakhstan, Moldova, Bosnia and Herzegovina,
+        // Slovenia, Albania, Montenegro, Georgia (using what was on https://openstreetmap.community/ )
```

Comment:
Replied regarding Slovenia in forum thread: https://forum.openstreetmap.org/viewtopic.php?pid=865336#p865336

```suggestion
        // Croatia (HR) - https://github.com/streetcomplete/StreetComplete/issues/3064#issuecomment-1157858833
        // Slovenia (SI) - https://forum.openstreetmap.org/viewtopic.php?pid=865336#p865336

        // was unable to reach communities or get replies from Kazakhstan, Moldova, Bosnia and Herzegovina,
        // Albania, Montenegro, Georgia (using what was on https://openstreetmap.community/ )```